kernel/src/analysis/tree_sitter.rs

```diff
@@ -129,12 +124,14 @@ pub fn map_node(node: tree_sitter::Node) -> Option<TreeSitterNode> {
                 line: u32::try_from(cursor.node().range().end_point.row + 1).unwrap(),
                 col: u32::try_from(cursor.node().range().end_point.column + 1).unwrap(),
             },
-            field_name: cursor.field_name().map(|v| v.to_string()),
+            field_name: cursor.field_name().map(ToString::to_string),
             children,
         };
 
         Some(ts_node)
     }
+
+    let mut ts_cursor = node.walk();
```

Comment:
Move this line after the internal function for readability's sake. See this [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/items_after_statements).